### PR TITLE
Update internal footer links to be language dynamic and fix Developer API link

### LINKED ISSUE #11233: Developer API footer link leads to homepage
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

Footer of the site.

### What changes are you suggesting?

Update footer link to point to developers page.

### Additional information

_No response_

### LINKED ISSUE #11274: Internal Footer links do not lead to selected translation page
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

Internals links in the footer are not dynamic with respect to language selected in the language picker.

### What changes are you suggesting?

Make the links to lead to respective selected translation page.

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by AnilSeervi**
### Why:

Closes #11233
Closes #11274 

### What's being changed:
- Update `Developer API` footer link to point to [developers](https://docs.github.com/en/developers) page rather than to homepage.
- Update internal footer links to be dynamic to user selected language.

### Check off the following:

- [x] I have reviewed my changes in staging (look for the latest deployment event in your pull request's timeline, then click **View deployment**).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by AnilSeervi** [comment]
@ramyaparimi I've got a question and I've posted it at https://github.com/github/docs/discussions/11235
Shouldn't the links in the footer too lead to corresponding pages of user-selected language.

**Comment by mikesurowiec** [comment]
> So..
> Am I allowed to use `location.pathname.split("/")[1]` to get the locale and then use it in the link 😅

In this case I'd prefer to `useRouter()` and get the locale from `router.locale`. There's an example in this file: https://github.com/github/docs/blob/main/components/page-header/LanguagePicker.tsx 🙏

**Comment by mikesurowiec** [comment]
> Am i missing something here or is the code wrong!!

I believe the behavior you're seeing on the generic error page is a separate bug 🙂. You can go ahead and useRouter in this case and I'll create a separate ticket internally to track the new issue

**Review by gracepark — APPROVED** [review]
Looks great! Thank you!

**Comment by AnilSeervi** [comment]
> Looks great! Thank you!

Yes. They're now working as expected :)

File: components/page-footer/SmallFooter.tsx [hunk 1]
```diff
@@ -1,21 +1,24 @@
...
+  const router = useRouter()
   const { t } = useTranslation('footer')
   return (
     <footer className="container-xl px-3 mt-6 mb-8 px-md-6 position-relative d-flex flex-row-reverse flex-xl-row flex-wrap flex-xl-nowrap flex-justify-center flex-xl-justify-between f6 color-text-secondary">
       <ul className="list-style-none d-flex flex-wrap col-12 col-xl-5 flex-justify-center flex-xl-justify-between mb-2 mb-xl-0">
         <li className="mr-3 mr-xl-0">&copy; {new Date().getFullYear()} GitHub, Inc.</li>
         <li className="mr-3 mr-xl-0">
-          <a href="https://docs.github.com/en/github/site-policy/github-terms-of-service">
+          <Link href={`/${router.locale}/github/site-policy/github-terms-of-service`}>
             {t('terms')}
```
[the remark this answers is not in the document]
Comment (peterbe) [reply components/page-footer/SmallFooter.tsx:15]: Is this one any different from the others? I'll become `<a href="/en/github/site-policy/github-terms-of-service">Terms</a>` when it renders.

[the remark this answers is not in the document]
Comment (AnilSeervi) [reply components/page-footer/SmallFooter.tsx:15]: Not particularly,
It's just the way *Nextjs* recommends using `<Link>` tag in their docs.

**Pull request merged**